src/app/components/client/FixNavigation.tsx

```diff
@@ -64,49 +69,79 @@ export const Steps = (props: {
     return value.length > 0;
   }).length;
 
+  const completedScan = hasCompletedStep(props.data, "Scan");
+  const completedHighRisk =
+    hasCompletedStep(props.data, "HighRiskSsn") &&
+    hasCompletedStep(props.data, "HighRiskCreditCard") &&
+    hasCompletedStep(props.data, "HighRiskBankAccount") &&
+    hasCompletedStep(props.data, "HighRiskPin");
+  const completedPasswords =
+    hasCompletedStep(props.data, "LeakedPasswordsPassword") &&
+    hasCompletedStep(props.data, "LeakedPasswordsSecurityQuestion");
+  const completedTips =
+    hasCompletedStep(props.data, "SecurityTipsEmail") &&
+    hasCompletedStep(props.data, "SecurityTipsIp") &&
+    hasCompletedStep(props.data, "SecurityTipsPhone");
+
   return (
     <ul className={styles.steps}>
-      <li
-        aria-current={
-          props.currentSection === "data-broker-profiles" ? "step" : undefined
-        }
-        className={`${styles.navigationItem} ${
-          props.currentSection === "data-broker-profiles" ? styles.active : ""
-        }`}
-      >
-        <div className={styles.stepIcon}>
-          <Image
-            src={stepDataBrokerProfilesIcon}
-            alt=""
-            width={22}
-            height={22}
-          />
-          {/* // TODO: Add logic to mark icon as checked when step is complete */}
-          {/* <CheckIcon className={styles.checkIcon} alt="" width={22} height={22} />  */}
-        </div>
+      {isEligibleForStep(props.data, "Scan") && (
+        <li
+          aria-current={
+            props.currentSection === "data-broker-profiles" ? "step" : undefined
+          }
+          className={`${styles.navigationItem} ${
+            props.currentSection === "data-broker-profiles" ? styles.active : ""
+          } ${completedScan ? styles.isCompleted : ""}`}
+        >
+          <div className={styles.stepIcon}>
+            {completedScan ? (
+              <CheckIcon
+                className={styles.checkIcon}
+                alt=""
+                width={22}
+                height={22}
+              />
+            ) : (
+              <Image
+                src={stepDataBrokerProfilesIcon}
+                alt=""
+                width={22}
+                height={22}
+              />
```

Comment:
Question: Is it worth it to make this its own component?

Reply:
Not sure if is, but I did it anyway :P Incorporated that in c4678a818 as well.